jaxtyping/_decorator.py

```diff
@@ -321,13 +322,12 @@ def __init__(self, *args, **kwargs):
                 # annotations as not needing instance checks, while still being
                 # visible as original ones for the typechecker
                 def modify_annotation(ann):
-                    if isinstance(ann, _MetaAbstractArray):
-                        setattr(ann, "_skip_instancecheck", True)
-                    elif hasattr(ann, "__args__"):
-                        for sub_ann in get_args(ann):
-                            modify_annotation(sub_ann)
-                    elif hasattr(ann, "__origin__"):
-                        modify_annotation(get_origin(ann))
+                    origin = ann if inspect.isclass(ann) else get_origin(ann)
+                    if origin is not None and issubclass(origin, AbstractArray):
+                        ann.make_transparent()
```

Comment:
Actually. do we ever need the origin check? I don't think we can ever have an `AbstractArray` as an origin. I think just `if inspect.isclass(ann) and `issubclass(ann, AbstractArray)` will work.

Side note, I managed to dig up an example of why explicitly checking `__origin__` or `__args__` should be avoided:
```python
x = Annotated[int, "foo"]
x.__origin__  # int
get_origin(x)  # Annotated
x.__args__  # (int,)
get_args(x)  # (int, 'foo')
```

Reply:
I think we do need the check, if ```ann``` is ```None```, then ```origin``` is ```None``` and ```issubclass``` fails. 

And we cannot just do something like ```origin = get_origin(ann); if inspect.isclass(ann) etc```, since then we are going to break if ```ann``` is already a class (I mean, already like an origin), which happens sometimes, and leads to ```origin``` being None.

Reply:
For example, ```ann``` can be a class when having something like ```f()->Union[int, float]```.

---
Also, your example with using the origin/args directly is very nice. I think I encountered something similar during the last week, but completely forgotten what it was :)

Reply:
```python
inspect.isclass(None)  # False
inspect.isclass(Union[int, str])  # False
```
So I think that's the only thing we need to check to be sure that `issubclass` is valid?

Reply:
Actually, you are right, I just checked. Well, well, another lesson for me: Patrick always right :)